code/game/objects/items/melee/misc.dm

```diff
@@ -326,6 +334,197 @@
+					return
+				if(check_martial_counter(H, user))
+					return
+
+			var/list/desc = get_stun_description(target, user)
+			var/obj/item/bodypart/La = target.get_bodypart(BODY_ZONE_L_ARM)
+			var/obj/item/bodypart/Ra = target.get_bodypart(BODY_ZONE_R_ARM)
+			var/obj/item/bodypart/Ll = target.get_bodypart(BODY_ZONE_L_LEG)
+			var/obj/item/bodypart/Rl = target.get_bodypart(BODY_ZONE_R_LEG)
+			var/mob/living/carbon/human/T = target
+
+			if (stun_animation)
+				user.do_attack_animation(target)
+			playsound(get_turf(src), on_stun_sound, 75, 1, -1)
+			additional_effects_carbon(target, user)
+			if((user.zone_selected == BODY_ZONE_CHEST))
+				target.apply_damage(stamina_damage, STAMINA, BODY_ZONE_CHEST, def_check)
+				log_combat(user, target, "stunned", src)
+				target.visible_message(desc["visiblestun"], desc["localstun"])
+
+			if((user.zone_selected == BODY_ZONE_HEAD))
+				target.apply_damage(12, STAMINA, BODY_ZONE_HEAD, def_check)
+
+				if(target.staminaloss > 89 && !target.has_status_effect(STATUS_EFFECT_SLEEPING) && (!sleep_cooldowns[target] || sleep_cooldowns[target] <= world.time))
+					T.force_say(user)
+					target.balloon_alert_to_viewers("Knock-Out!")
+					if(!target.has_status_effect(STATUS_EFFECT_SLEEPING))
+						target.Sleeping(80)
+						target.setStaminaLoss(0)
+						playsound(usr.loc, "sound/machines/bellsound.ogg", 15, 1)
+						log_combat(user, target, "Knocked-Out", src)
+					if(CHECK_BITFIELD(target.mobility_flags, MOBILITY_STAND)) //this is here so the "falls" message doesn't appear if the target is already on the floor
+						target.visible_message("<span class='emote'><b>[T]</b> [pick(list("falls unconscious.","falls limp like a bag of bricks.","falls to the ground, unresponsive.","lays down on the ground for a little nap.","got [T.p_their()] dome rung in."))]</span>")
+					else
+						target.visible_message("<span class='emote'><b>[T]</b> [pick(list("falls unconscious.","falls into a deep sleep.","was sent to dreamland.","closes [T.p_their()] and prepares for a little nap."))]</span>")
+					sleep_cooldowns[target] = world.time + 16 SECONDS
+				else
+					log_combat(user, target, "stunned", src)
+					target.visible_message(desc["visiblestun"], desc["localstun"])
+
+			if(user.zone_selected == BODY_ZONE_L_LEG)
+				log_combat(user, target, "stunned", src)
+				target.visible_message(desc["visibleleg"], desc["localleg"])
+				if (Rl.get_staminaloss() < 26 && Ra.get_staminaloss() < 26 && La.get_staminaloss() < 26)
+					target.apply_damage(15, STAMINA, BODY_ZONE_L_LEG, def_check)
+				else
+					target.apply_damage(10, STAMINA, BODY_ZONE_L_LEG, def_check)
+				if (Ll.get_staminaloss() == 50)
+					target.apply_damage(8, STAMINA, BODY_ZONE_CHEST, def_check)
+				else
+					target.apply_damage(4, STAMINA, BODY_ZONE_CHEST, def_check)
+
+				if(Ll.get_staminaloss() == 50 && CHECK_BITFIELD(target.mobility_flags, MOBILITY_STAND) && (!trip_cooldowns[target] || trip_cooldowns[target] <= world.time))
```

Comment:
```suggestion
				if(Ll.get_staminaloss() == 50 && CHECK_BITFIELD(target.mobility_flags, MOBILITY_STAND) && (!trip_cooldowns[target] || COOLDOWN_FINISHED(src, trip_cooldowns[target]))
```

Reply:
@Absolucy did this part of the code so she'd be better equip to answer this one!

Reply:
Honestly I'd never thought of using the macros on lists haha